src/System.CommandLine/Symbol.cs

```diff
@@ -54,6 +72,8 @@ internal void AddParent(Symbol symbol)
                 }
                 current.Next = new ParentNode(symbol);
             }
+
+            static void Throw(Symbol symbol) => throw new ArgumentException($"Cycle detected in symbol tree. Symbol '{symbol.Name}' would be its own ancestor.");
```

Comment:
The message should maybe say "Command ____ would be its own ancestor" as it should be impossible for this to happen with other symbol types.